### PR TITLE
Use new table of CSET_id to unnormalized org name linkages

### LINKED ISSUE #310: Integrate new unnormalized name to ROR linkages
See https://github.com/georgetown-cset/parat/pull/304

### PR DESCRIPTION
**Pull request by jmelot**
Closes #310 

Still needed:

- [ ] Implementing caching of raw name to CSET id matches
- [ ] Running all queries and carefully checking results relative to old values
- [ ] Probably some schema and downstream code updates (I changed a few column names to make them more consistent with each other)

@rggelles this is the continuation of https://github.com/georgetown-cset/parat/pull/304. It still needs some polish, but I think this is a good time to start to get your input. The basic idea of what I'm doing here is replacing `get_ai_counts.py` and its callers with a series of sql queries that use the new `org_name_matches` table to link org names to CSET ids.

I want to make sure this method makes sense to both of us. If you're willing, I think a good place to start would be understanding this count discrepancy: https://github.com/georgetown-cset/parat/pull/315/files#r1587841664

If you could just take a look at that query and confirm that the method makes sense to you, I can run the pipeline up to the production copy stage and come back with a longer writeup on diffs between old and new counts.

### CODE REVIEW

File: company_linkage/sql/ai_company_papers.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
+WITH unnested_rors AS (
```
Comment (jmelot) [inline company_linkage/sql/ai_company_papers.sql:1]: @rggelles this query shows how the new `org_name_matches` table is used. One mystery is that it returns only `89,643 rows`, as opposed to the 130,154 in `staging_ai_companies_visualization.ai_company_papers`. A difference between the old and new methods is that the [new method](https://github.com/georgetown-cset/parat/blob/dee2515a233012be30030de7aa1c9518b516f8c4/company_linkage/sql/org_name_cross_regex.sql) only searches rows where the ror id is null. Do you see any other differences that might account for this?

**Review by rggelles — COMMENTED** [review]
The method here makes sense to me. I haven't reviewed the whole thing yet, but from what I see, as long as we're okay with the slight shift in methodology (which is really a reversion to our previous methodology) then there's no problems here.

File: company_linkage/sql/ai_company_papers.sql (hunk 1 above)
Comment (jmelot) [reply company_linkage/sql/ai_company_papers.sql:1]: I implemented the fix to regex case sensitivity as discussed last week but this issue persists in a less extreme form - about to post a follow-up below

**Comment by rggelles** [comment]
So, to clarify here: we fully annotated every single ROR organization in PARAT.

The gaps you are seeing here are because it has been 1.5 years since our annotation efforts completed, and ROR has added new organizations in that time. For example, Hugging Face did not have a ROR when we annotated; according to ROR it looks most likely it was created 2022-10-27, which is a month or two after we had wrapped up our annotation. I honestly have no idea what could have gone wrong with Synopsys, because it absolutely has GRIDs (4!) in the annotation table, and those GRIDs are linked to ROR in the ROR table, so I have no idea why it would not have ROR in high_resolution_entities; some error is happening here, but it's not an annotation problem.

I don't know if there's an easy way from your end to see the organizations whose paper totals dropped the most between the two versions, but if so, it seems like the easiest solution would be to see if there's a small set of those and if those are organizations that acquired ROR in the past 1.5 years (or had whatever weird linking problem Synopsys is having, which we should get fixed). Then we could add RORs for just those, and solve this. I doubt the number is that high? But I suppose I could be wrong.

Otherwise, we could just switch to running over everything, if this is too difficult or time-consuming.

**Comment by jmelot** [comment]
Interesting. I just did `select * from high_resolution_entities.aggregated_organizations where array_length(ror_id) = 0` and three of the first 10 records that popped up for me (e.g https://ror.org/04g9yb321) had ROR ids circa 2021. The other 7 weren't obviously in ror (i.e. I didn't immediately see a matching org when I searched in the ROR website). We should submit all missing PARAT orgs to ROR, maybe after PARAT v2 finally gets out the door. I'm familiar with the process for doing this in bulk from doing it with James, Daniel, and Zach on another dataset so happy to do it, or support if you'd like to do it.

Regardless, I think we should either commit to a process that matches each org to ROR ids in the near-term and going forward, or switch back to doing regular expression search over all orgs regardless of whether they have ROR ids. 

I assume you're not able to support further ROR matching, so I will discuss that with Zach. In the meantime, would you be able to take a look at why we're failing to match some of these orgs that were apparently in ROR in 2021, just in case it's symptomatic of some other issue? Once I confirm that we're not going to have comprehensive ROR matches in the near term I'll switch the query over to matching all orgs and can ping you again to review once the data is updated.

**Comment by za158** [comment]
Just noting that many PARAT orgs will have 0 research publications, so it
makes sense that they lack ROR/GRID - and for the same reason I'm not sure
ROR will take them...
---
Zachary Arnold
Analytic Lead, Emerging Technology Observatory (web <http://eto.tech> |
LinkedIn <https://www.linkedin.com/company/emerging-technology-observatory>
| Substack <http://etoblog.substack.com>)
Center for Security and Emerging Technology
***@***.***

**Comment by jmelot** [comment]
Anyway, Zach's annotators haven't been adding RORs, so I think we should just fall back to doing regex search for all orgs if we don't have a way of keeping the ROR ids up to date right now. I do think it would still be a good thing to figure out why Synopsys etc don't have RORs though, mostly to make sure it isn't symptomatic of something else. But if you approve @rggelles I'll just switch over to doing regex search on everything

**Comment by jmelot** [comment]
Gotcha - but fyi Zach had the good idea to just count the number of orgs with pubs but no rors and it's 845 from `SELECT count(0) FROM ai_companies_visualization.all_visualization_data where array_length(ror_id) = 0 and all_pubs > 0`

**Comment by jmelot** [comment]
That's presumably a combination of the ROR circa 1.5 years ago coverage issue and the Synopsis issue in some unknown proportions. Thanks for looking at the Synopsis issue. In the meantime, I'll switch over to doing regex searches for all orgs

**Comment by rggelles** [comment]
Okay, to fix the Synopsys issue, what needs to be done is the following:

Go [here](https://github.com/georgetown-cset/parat/blob/b7765ec441f8842958c7e976c9595683e2f48257/company_linkage/sql/organizations.sql#L109) and add a TRIM() around `grid_joined.grid`.

This will solve the fact that some of the GRID annotation fields apparently must have ended up with spaces in them 🤦🏻‍♀️

**Comment by jmelot** [comment]
Ok thanks - I made that change and also run regex on everything now. `select count(0) from staging_ai_companies_visualization.ai_company_papers` = 142,992 which makes more sense since the number of papers increased since this was last run. @rggelles can you take the review from here?

**Review by rggelles — APPROVED** [review]
LGTM!

**Pull request merged**